packages/dev/core/src/assetContainer.ts

```diff
@@ -1050,4 +1052,97 @@ export class AssetContainer extends AbstractScene {
             }
         });
     }
+
+    private _checkAndAddTextureStandardMaterial(material: StandardMaterial, textureName: keyof StandardMaterial) {
+        if (material[textureName]) {
+            this.textures.push(material[textureName]);
+        }
+    }
+
+    private _checkAndAddTexturePBRMaterial(material: PBRMaterial, textureName: keyof PBRMaterial) {
+        if (material[textureName]) {
+            this.textures.push(material[textureName]);
+        }
+    }
+
+    /**
+     * @since
+     * Given a root asset, this method will traverse its hierarchy and add it, its children and any materials/skeletons/animation groups to the container.
+     * @param root
+     */
+    public addAllAssetsToContainer(root: Node) {
+        if (!root) {
+            return;
+        }
+
+        const nodesToVisit = new Array<Node>();
+        const visitedNodes = new Set<Node>();
+
+        nodesToVisit.push(root);
+
+        while (nodesToVisit.length > 0) {
+            const nodeToVisit = nodesToVisit.pop()!;
+            visitedNodes.add(nodeToVisit);
+
+            if (nodeToVisit instanceof Mesh) {
+                if (nodeToVisit.geometry) {
+                    this.geometries.push(nodeToVisit.geometry);
+                }
+                this.meshes.push(nodeToVisit);
+            } else if (nodeToVisit instanceof TransformNode) {
+                this.transformNodes.push(nodeToVisit);
+            } else if (nodeToVisit instanceof Light) {
+                this.lights.push(nodeToVisit);
+            } else if (nodeToVisit instanceof Camera) {
+                this.cameras.push(nodeToVisit);
+            }
+
+            if (nodeToVisit instanceof AbstractMesh) {
+                if (nodeToVisit.material) {
+                    this.materials.push(nodeToVisit.material);
+                    if (nodeToVisit.material instanceof StandardMaterial) {
+                        this._checkAndAddTextureStandardMaterial(nodeToVisit.material, "diffuseTexture");

```

Comment:
could we use getActiveTextures instead ?

Reply:
does getActiveTextures encompass node material texture blocks as well?

Reply:
> does getActiveTextures encompass node material texture blocks as well?

It does 🙂